### PR TITLE
Make documentation explicit on the relation between roles and permissions

### PR DESCRIPTION
**Pull request by pmlopes**
This is a sum up of the discussions on IRC and extra examples for matching with wildcards

Relates to #16.

### CODE REVIEW

**Comment by pmlopes** [comment]
With the 3 commits the refactor is complete.

File: vertx-auth-common/src/main/java/io/vertx/ext/auth/package-info.java [hunk 1]
```diff
@@ -72,10 +74,8 @@
  *
  * Once you have an {@link io.vertx.ext.auth.User} instance you can call methods on it to authorise it.
  *
- * To check if a user has a specific role you use {@link io.vertx.ext.auth.User#hasRole},
- * to check if a user has all the specified roles you use {@link io.vertx.ext.auth.User#hasRoles},
- * to check if a user has a specific permission you use {@link io.vertx.ext.auth.User#hasPermission},
- * to check if a user has all the specified permissions you use {@link io.vertx.ext.auth.User#hasPermissions}.
+ * to check if a user has a specific permission you use {@link io.vertx.ext.auth.User#isPermitted},
+ * to check if a user has all the specified permissions you use {@link io.vertx.ext.auth.User#isPermitted}.
```
Comment (purplefox) [inline vertx-auth-common/src/main/java/io/vertx/ext/auth/package-info.java]: this should be removed

File: vertx-auth-common/src/main/java/io/vertx/ext/auth/package-info.java [hunk 2]
```diff
@@ -86,10 +86,13 @@
  * {@link examples.Examples#example2}
  * ----
  *
- * === Caching roles and permissions
+ * Matching of Permissions is expected to be done using String equality unless otherwise stated by the underlying
+ * implementation.
```
Comment (purplefox) [inline vertx-auth-common/src/main/java/io/vertx/ext/auth/package-info.java]: We should probably say something that the actual meaning of the permissions is up to the provider. I don't think strict equality is required.

File: vertx-auth-jdbc/src/main/java/io/vertx/ext/auth/jdbc/JDBCAuth.java [hunk 3]
```diff
@@ -33,17 +33,12 @@
...
   /**
-   * The default query to retrieve all roles for the user
+   * The default query to retrieve all permissions for the user
    */
-  static final String DEFAULT_ROLES_QUERY = "SELECT ROLE FROM USER_ROLES WHERE USERNAME = ?";
-
-  /**
-   * The default query to retrieve all permissions for the role
-   */
-  static final String DEFAULT_PERMISSIONS_QUERY = "SELECT PERM FROM ROLES_PERMS RP, USER_ROLES UR WHERE UR.USERNAME = ? AND UR.ROLE = RP.ROLE";
```
Comment (purplefox) [inline vertx-auth-jdbc/src/main/java/io/vertx/ext/auth/jdbc/JDBCAuth.java:46]: Shouldn't we preserve the previous functionality and use a prefix "role:" to signify role. I think in practice most users using the JDBC auth will be using a user/role/permission model.

File: vertx-auth-shiro/src/test/java/io/vertx/ext/auth/test/shiro/PropertiesShiroAuthProviderTest.java [hunk 4]
```diff
@@ -38,4 +39,31 @@ protected JsonObject getConfig() {
     return config;
   }
 
+  @Test
```
Comment (purplefox) [inline vertx-auth-shiro/src/test/java/io/vertx/ext/auth/test/shiro/PropertiesShiroAuthProviderTest.java:42]: I think we still need the roles tests...

**Comment by pmlopes** [comment]
docs and support for roles is restored on JDBC still need to document this and make the prefix a config option.

**Comment by pmlopes** [comment]
So current PR is now complete i think. We have streamlined the API with just: `isPermitted` and for providers that support the concept of roles there is prefix `role:` which is configurable we can run assertions like: `isPermitted("role:admin")`.

The role support is implemented for:
- jdbc
- shiro

The docs have been updated to reflect this change.

**Pull request merged**